File: Data Structures & Algorithms/two-integer-sum/submission-3.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)

    for i, num := range nums{
        m[num] = i
    }

    for i, num := range nums{
        diff := target - num
        if idx, ok := m[diff]; ok && idx != i{
            return []int{i, idx}
        }
    }

    return []int{}
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int idx = 0;
        for(int i =1;i<nums.size();i++){
            if(nums[idx] != nums[i]){
                nums[idx+1] = nums[i];
                idx++;
            }
        }
        return idx+1;
    }
}; 